### ISSUE TITLE
How to mqtt export to AWS IoT endpoint

### ISSUE THREAD

**Issue by Vieufoux** [edgexfoundry/app-service-configurable#209]
Hi I can send EdgeXEvent to HiveMQ endpoint success, but i cann't find any configuration about how to send event to AWS IoT endpoint.

Here is my configuration in my docker-compose.yaml

`  app-service-mqtt:
    container_name: edgex-app-service-configurable-mqtt
    depends_on:
    - consul
    - data
    environment:
      BINDING_PUBLISHTOPIC: events
      CLIENTS_COMMAND_HOST: edgex-core-command
      CLIENTS_COREDATA_HOST: edgex-core-data
      CLIENTS_DATA_HOST: edgex-core-data
      CLIENTS_METADATA_HOST: edgex-core-metadata
      CLIENTS_NOTIFICATIONS_HOST: edgex-support-notifications
      CLIENTS_RULESENGINE_HOST: edgex-kuiper
      CLIENTS_SCHEDULER_HOST: edgex-support-scheduler
      CLIENTS_VIRTUALDEVICE_HOST: edgex-device-virtual
      DATABASES_PRIMARY_HOST: edgex-redis
      EDGEX_PROFILE: mqtt-export
      EDGEX_SECURITY_SECRET_STORE: "false"
      MESSAGEBUS_SUBSCRIBEHOST_HOST: edgex-core-data
      REGISTRY_HOST: edgex-core-consul
      SERVICE_HOST: edgex-app-service-configurable-mqtt
      SERVICE_PORT: 48101
      Writable_Pipeline_Functions_MQTTSend_Addressable_Address: aaaaaaaaaaaa.iot.eu-west-1.amazonaws.com
      Writable_Pipeline_Functions_MQTTSend_Addressable_Port: 8883
      Writable_Pipeline_Functions_MQTTSend_Addressable_Protocol: tcp
      Writable_Pipeline_Functions_MQTTSend_Addressable_Topic: TestTopic
      WRITABLE_PIPELINE_FUNCTIONS_MQTTSEND_PARAMETERS_KEY: /tmp/123456-private.pem.key
      WRITABLE_PIPELINE_FUNCTIONS_MQTTSEND_PARAMETERS_CERT: /tmp/123456-certificate.pem.crt
    hostname: edgex-app-service-configurable-mqtt
    image: edgexfoundry/docker-app-service-configurable:1.3.0
    networks:
      edgex-network: {}
    ports:
    - 48101:48101/tcp
    read_only: true
`

**Comment by lenny-intel** [comment]
What issues are you seeing?

A few things jump out to me. 

1. You specify `Protocol` to be `tcp` when you are using port `8883` which I think requires `tcps`
2. I don't see any volume mount for the `key` & `cert` files

Also note that `MQTTSend` has been deprecated and replaced by `MqttSecretSend`
https://docs.edgexfoundry.org/1.3/microservices/application/AppServiceConfigurable/#mqttsecretsend
 `MQTTSend` has been removed in the upcoming 2.0 release.
`MqttSecretSend` looks for the `key` & `cert` in the SecretStore rather than a file. You can use the new InsecureSecrets config section for your secret store. The config exists here for overrides in the compose file.
https://github.com/edgexfoundry/app-service-configurable/blob/hanoi/res/mqtt-export/configuration.toml#L64
Example overrides for :
```
	WRITABLE_INSECURESECRETS_MQTT_SECRETS_CLIENTCERT: <your cert content>
	WRITABLE_INSECURESECRETS_MQTT_SECRETS_CLIENTKEY: <your key content>
```
I recommend to review the documentation for `MqttSecretSend` and switch to using it.

Also note we have released 1.3.1 for Hanoi. The released compose files have been updated with the 1.3.1 version of the images.
https://wiki.edgexfoundry.org/pages/viewpage.action?pageId=58294439

**Comment by lenny-intel** [comment]
Closing issue as answered
